### PR TITLE
Add routing algorithm and automation testing

### PR DESCRIPTION
**Pull request by phillipet**
- add new routing algorithm for customer and admin user
- add test cases routing (no-token, customer token, admin token)
- add automation test function for all test cases (test cafe framework)

### CODE REVIEW

File: test/Routing.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import {ClientFunction} from 'testcafe'
+const setLocalStorageItem = ClientFunction((prop, value) => {
+  window['ResizeObserver'] = undefined
```
Comment (hemvanh) [inline test/Routing.ts:4]: add comment: 
_**// temp fix for "ResizeObserver loop limit exceeded" in Chrome**_
let token = localStorage.getItem('auth-token')

return token ? token.substr(token.length - 1) : null

File: src/router/index.js [hunk 2]
```diff
@@ -22,23 +22,72 @@ export default function({store}) {
   })
 
   Router.beforeEach((to, _, next) => {
-    // routing page by authenticate token
-    if (isAuth()) {
-      if (to.path === '/customer') {
+    let userType = getUserType()
```
Comment (hemvanh) [inline src/router/index.js]: switch(getUserType()){
  case 'a':
  case 'c':
  default:

File: src/router/index.js [hunk 3]
```diff
@@ -22,23 +22,72 @@ export default function({store}) {
...
+    // user is CUSTOMER
+    if (userType === 'c') {
+      // route to customer home page
+      if (to.path === '/customer' || to.path === '/customer/') {
```
Comment (hemvanh) [inline src/router/index.js]: let toPath=to.path
...
use switch(toPath) instead of lots of if.. else if.. 
--> easier to read and maintain later

**Review by hemvanh — CHANGES_REQUESTED** [review]
fine tune the code abit more

**Comment by phillipet** [comment]
I changed

**Review by hemvanh — APPROVED** [review]

**Pull request merged**